### PR TITLE
Add support for Leo premium

### PR DESCRIPTION
**Pull request by nvonpentz**
Resolves https://github.com/brave/brave-browser/issues/32738
Resolves https://github.com/brave/brave-browser/issues/33143
Serverside PR https://github.com/brave/aichat/pull/36
Security review https://github.com/brave/reviews/issues/1424

Integrates with the SKUs SDK to:
1. Expose an API `GetPremiumStatus` to frontends that returns whether there are valid premium credentials available.  Uses [CredentialSummary](https://github.com/brave/brave-core/blob/3da0ac5ffefe0e34306450b521afb85ec78f7b27/components/skus/common/skus_sdk.mojom#L12C3-L12C20) from the SKUs SDK to determine if there are credentials available.
2. Fetch credentials to be included in the Cookie header in http requests to our premium AI chat endpoint.  These credentials will be validated in our server code https://github.com/brave/aichat/pull/36. Uses [PrepareCredentialsPreparation](https://github.com/brave/brave-core/blob/3da0ac5ffefe0e34306450b521afb85ec78f7b27/components/skus/common/skus_sdk.mojom#L11) from the SKUs SDK.

Most of this logic has is handled by a new AIChatCredentialManager class.

Because the call to the SKUs sdk `PrepareCredentialsPresentation` actually consumes the credential, we must be careful to save credentials retrieved from the SDK but not successfully used in an http request.  If the server is down or the user has network issues, they could use all of them without actually making a successful call.  To handle this, I added logic that checks for a valid credential stored in local state prefs before making a call to the SKU SDK, and puts credentials back into the cache upon server error or request timeout.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Create an account on account.bravesoftware.com (creds in 1password), purchase leo subscription. You can use 424242... (repeating) for the Stripe credit card
2. Start the browser using the dev / staging environments (some parts of this run on staging, some on dev) `npm run start -- --enable-features=AIChat --env-leo=staging --env-ai-chat.bsg=dev --env-ai-chat-premium.bsg=dev`
3. If that part of the integration worked, you should see your credentials in brave://skus-internals, and you will use those credentials for API access by default
4. To verify the premium credentials are actually being used, you may need to add some logging

### CODE REVIEW

File: browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc [hunk 1]
```diff
@@ -290,4 +291,9 @@ void AIChatUIPageHandler::OnVisibilityChanged(content::Visibility visibility) {
   active_chat_tab_helper_->OnConversationActiveChanged(is_visible);
 }
 
+void AIChatUIPageHandler::UserHasValidPremiumCredential(
+    UserHasValidPremiumCredentialCallback callback) {
+  active_chat_tab_helper_->UserHasValidPremiumCredential(std::move(callback));
```
Comment (petemill) [inline browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc:296]: I think we could skip AIChatTabHelper, and call `UserHasValidPremiumCredential` on the credential manager here, more directly. What do you think?

Comment (nvonpentz) [reply browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc:296]: I considered that, but ran into issues. The problem is that I need to eventually pass the AIChatCredentialManager when initializing the RemoteCompletionClient inside the AIChatTabHelper, and the AIChatTabHelper doesn't appear to be initialized by AIChatUIPageHandler, but instead in [browser/brave_tab_helpers.cc](https://github.com/brave/brave-core/blob/eb7d8ed8985a1a8c7dd3d58399ac4144095dd70c/browser/brave_tab_helpers.cc#L123-L126).

So I wouldn't be able to pass the page handler's instance of the credential manager through to the remote completion client. The way I'm understanding it, I'd need to create two instances of the credential manager, one for the page helper, and one for the tab helper (which may actually be fine).

I could also be missing something -- I'm not super familiar with tab helpers.

File: browser/brave_tab_helpers.cc [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #include "brave/browser/misc_metrics/page_metrics_tab_helper.h"
 #include "brave/browser/misc_metrics/process_misc_metrics.h"
 #include "brave/browser/ntp_background/ntp_tab_helper.h"
+#include "brave/browser/skus/skus_service_factory.h"
```
Comment (petemill) [inline browser/brave_tab_helpers.cc:22]: this doesn't seem needed

File: browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc (hunk 1 above)
Comment (petemill) [reply browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc:296]: I see. It seems to me that we only need one CredentialManager for a profile. In which case, the CredentialManager could be a profile keyed service like  `SkusService` is (it would be `AIChatCredentialService`. Although, when we have a `AIChatService` in the near future, we could just put the `AIChatCredentialManager` instance there instead of each TabHelper. So maybe this is fine for now. It's just a bit wasteful as we'll have an instance per-Tab somewhat unneccessarily.

Comment (nvonpentz) [reply browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc:296]: I think I prefer putting the credential manager with AIChatService if it's tolerable for the time being to have per tab instances

File: components/ai_chat/browser/constants.h [hunk 3]
```diff
@@ -17,6 +17,8 @@ base::span<const webui::LocalizedString> GetLocalizedStrings();
 
 bool UsesLlama2PromptTemplate(const std::string& model);
 
+extern const char kLeoSkuDomain[];
```
Comment (petemill) [inline components/ai_chat/browser/constants.h:20]: If this is only used in 1 file (`ai_chat_crendential_manager.cc`) then it can be defined at the top of that file, in an anonymous namespace.

File: components/ai_chat/browser/ai_chat_credential_manager_unittest.cc [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+    shared_url_loader_factory_ =
+        base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+            &url_loader_factory_);
+    skus_service_ = std::make_unique<skus::SkusServiceImpl>(
+        &prefs_service_, url_loader_factory_.GetSafeWeakWrapper());
+
+    ai_chat_credential_manager_ = std::make_unique<AIChatCredentialManager>(
+        base::BindRepeating(&AIChatCredentialManagerUnitTest::GetSkusService,
+                            base::Unretained(this)),
```
[the remark this answers is not in the document]
Comment (thypon) [reply components/ai_chat/browser/ai_chat_credential_manager_unittest.cc:47]: test

**Comment by petemill** [comment]
@nvonpentz do you have a test plan - a way in dev to get credentials?

File: components/ai_chat/common/mojom/ai_chat.mojom [hunk 5]
```diff
@@ -63,6 +63,7 @@ interface PageHandler {
   ClearConversationHistory();
   RetryAPIRequest();
   GetAPIResponseError() => (APIError error);
+  UserHasValidPremiumCredential() => (bool result);
```
Comment (petemill) [inline components/ai_chat/common/mojom/ai_chat.mojom:66]: nit: `GetHasValidPremiumCredential()` (or just `GetIsPremium()` unless we're also planning to add something similar to distinguis "is premium" vs "is premium with valid credential"

File: components/ai_chat/common/mojom/ai_chat.mojom [hunk 6]
```diff
@@ -27,6 +27,12 @@ enum APIError {
   RateLimitReached
 };
 
+enum PremiumStatus {
+  Inactive,
+  Active,
+  Disconnected
```
Comment (petemill) [inline components/ai_chat/common/mojom/ai_chat.mojom:33]: It's not clear which one means "Is not a subscribed premium user", so maybe:
```
IsPremium
IsNotPremium
IsPremiumDisconnected
```

any thoughts?

Comment (nvonpentz) [reply components/ai_chat/common/mojom/ai_chat.mojom]: The "Is" prefix feels more appropriate for the name of a boolean variable. 

How about:

```
Subscribed
NotSubscribed or Unsubscribed
SubscribedDisconnected
```
I'd also replace Subscribed with 'Active' / 'Inactive'.

If we want to use Premium / NotPremium / PremiumDisconnected, then I'd rename the 'PremiumStatus' to 'SubscriptionStatus'.

Any of these work for me.

Comment (petemill) [reply components/ai_chat/common/mojom/ai_chat.mojom]: ok let's go with `Active` | `Inactive` | `ActiveDisconnected` then, as it's the most minimal change but clears it up for me - thanks

File: build/commands/lib/config.js [hunk 7]
```diff
@@ -221,6 +221,7 @@ const Config = function () {
   this.goma_offline = false
   this.use_libfuzzer = false
   this.brave_ai_chat_endpoint = getNPMConfig(['brave_ai_chat_endpoint']) || ''
+  this.brave_ai_chat_premium_endpoint = getNPMConfig(['brave_ai_chat_premium_endpoint']) || ''
```
Comment (petemill) [inline build/commands/lib/config.js:224]: We won't need to add these buildflags as once https://github.com/brave/brave-core/pull/20249 is merged we can just prepend to the `brave_services_production_domain` build flag

File: browser/brave_tab_helpers.cc (hunk 2 above)
Comment (nvonpentz) [reply browser/brave_tab_helpers.cc:22]: It wasn't originally, but I have since shifted the code around, and it is now.

File: components/skus/browser/rs/lib/src/sdk/credentials/mod.rs [hunk 8]
```diff
@@ -182,14 +182,29 @@ where
...
+                .collect();
+            orders.sort_by(|a, b| b.expires_at.cmp(&a.expires_at));
+            if let Some(order) = orders.first() {
+                // We have at least one order for the specified location
+                if let Ok(Some(summary)) =
+                    self.matching_order_credential_summary(&order.id, domain).await
+                {
+                    return Ok(Some(summary));
+                } else {
+                    if let Some(expires_at) = order.expires_at {
```
Comment (rillian) [inline components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:197]: Thanks for adding a comment. In general this code would benefit from a few more. 

It looks like if there's a credential matching the domain and a summary can be created, it's returned. If not, and the first credential happens to be unexpired, a summary is created for it directly. What is the intent here?

`matching_order_credential_summary` refreshes the credentials for the order if they're expired. Why is that not necessary here? It also checks for domain match again, so why not just call it up front, before sorting by expiry time? Would that be a performance issue?

**Review by rillian — APPROVED** [review]
rust++ although I'm finding the intent here hard to follow. See below.

File: components/skus/browser/rs/lib/src/sdk/credentials/mod.rs (hunk 8 above)
Comment (nvonpentz) [reply components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:197]: The high level motivation for this change was to give callers of the CredentialSummary in the SKU SDK a way to differentiate between having an inactive subscription vs. having an active subscription with only expired credentials (requiring users to log in to account.brave.com to refresh).  Previously, CredentialSummary was returning `{}` for both of those case.  Now, the CredentialSummary returns a non empty object in the second case that has `remaining_credential_count: 0` as a key and value.

As far as the specifics of the code, I'm not as familiar. Maybe @evq can comment as they provided me this patch as well.

Comment (rillian) [reply components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:197]: Ok, if the `remaining_credential_count` is the important point, that makes more sense. Thanks for clarifying; feel free to document that in the actual code. :)

File: components/ai_chat/core/engine/remote_completion_client.cc [hunk 9]
```diff
@@ -94,19 +95,15 @@ std::string CreateJSONRequestBody(base::ValueView node) {
...
-  }
+const GURL GetEndpointBaseUrl(bool premium) {
+  auto* endpoint_suffix = premium ? "-premium.bsg" : ".bsg";
+  auto endpoint = brave_domains::GetServicesDomain(
+      base::StrCat({kAIChatHostnamePart, endpoint_suffix}));
+
+  static base::NoDestructor<GURL> url{base::StrCat(
+      {url::kHttpsScheme, url::kStandardSchemeSeparator, endpoint})};
+
+  return *url;
```
Comment (petemill) [inline components/ai_chat/core/engine/remote_completion_client.cc:106]: IMO this is confusing to have part of the subdomain as a constant in a different part of the file, and part of it here. Perhaps either put it as 2 constants at the top of the file, or just put the "ai-chat" part here with the rest of the subdomain.

(also, nit: the name endpoint doesn't make sense as it implies it's the full url)

```suggestion
  auto* prefix = premium ? "ai-chat-premium.bsg" : "ai-chat.bsg";
  auto hostname = brave_domains::GetServicesDomain(prefix);

  static base::NoDestructor<GURL> url{base::StrCat(
      {url::kHttpsScheme, url::kStandardSchemeSeparator, hostname})};

  return *url;
```

File: components/ai_chat/core/engine/remote_completion_client.cc [hunk 10]
```diff
@@ -116,17 +113,26 @@ const GURL GetEndpointBaseUrl() {
...
+    CHECK(api_base_url.is_valid()) << "Default API Url generated was invalid."
+                                      "Please check configuration parameter.";
+  }
+
+  api_base_url = GetEndpointBaseUrl(true);
   if (!api_base_url.is_empty()) {
     // Crash quickly if we have an invalid non-empty Url configured
     // as a build flag
-    CHECK(api_base_url.is_valid()) << "API Url generated was invalid."
+    CHECK(api_base_url.is_valid()) << "Premium API Url generated was invalid."
```
Comment (petemill) [inline components/ai_chat/core/engine/remote_completion_client.cc:135]: I don't think we need these checks on startup anymore - the URL will always be valid.

File: components/skus/browser/rs/lib/src/sdk/credentials/mod.rs (hunk 8 above)
Comment (nvonpentz) [reply components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:197]: cc @bsclifton since this will change the behavior of the SKU SDK, potentially impacting it's use for VPN.  Based on my reading of this [code](https://github.com/brave/brave-core/blob/f54931821e3dd5cddace52b9a363190a5038c9b5/components/brave_vpn/browser/brave_vpn_service_helper.cc#L44-L50), though it seems you are already expecting the SDK to behave in the way I've updated it

File: components/ai_chat/core/ai_chat_credential_manager.h [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+// Interfaces with the SKUs SDK to provide APIs to check and fetch Leo
+// premium credentials.
+class AIChatCredentialManager {
+ public:
+  AIChatCredentialManager(
+      base::RepeatingCallback<mojo::PendingRemote<skus::mojom::SkusService>()>
+          skus_service_getter,
+      PrefService* prefs_service);
```
Comment (nvonpentz) [inline components/ai_chat/core/ai_chat_credential_manager.h:35]: Note: This uses local state prefs, not profile prefs, which is consistent with BraveVPNService.

File: components/ai_chat/common/pref_names.h [hunk 12]
```diff
@@ -20,11 +20,15 @@ constexpr char kBraveChatP3AChatCountWeeklyStorage[] =
     "brave.ai_chat.p3a_chat_count";
 constexpr char kBraveChatP3APromptCountWeeklyStorage[] =
     "brave.ai_chat.p3a_prompt_count";
+constexpr char kBraveChatPremiumCredentialCache[] =
```
Comment (bridiver) [inline components/ai_chat/common/pref_names.h:23]: please add comments explaining what this is

File: test/BUILD.gn [hunk 13]
```diff
@@ -361,8 +361,19 @@ test("brave_unit_tests") {
   }
 
   if (enable_ai_chat) {
-    sources += [ "//brave/components/ai_chat/core/ai_chat_metrics_unittest.cc" ]
-    deps += [ "//brave/components/ai_chat/core" ]
+    sources += [
+      "//brave/components/ai_chat/core/ai_chat_credential_manager_unittest.cc",
```
Comment (bridiver) [inline test/BUILD.gn:365]: please do not add individual test sources to `test/BUILD.gn`. You should add a test target in the component and add a dep here for it (and remove the other deps)

File: browser/brave_local_state_prefs.cc [hunk 14]
```diff
@@ -136,6 +140,10 @@ void RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
   brave_vpn::RegisterLocalStatePrefs(registry);
 #endif
 
+#if BUILDFLAG(ENABLE_AI_CHAT)
+  ai_chat::prefs::RegisterLocalStatePrefs(registry);
```
[the remark this answers is not in the document]
Comment (bridiver) [reply browser/brave_local_state_prefs.cc:144]: Looking more at the code, this seems wrong. Skus is a browser context keyed service, but you're storing the credentials in local state so they would be shared across all profiles.

[the remark this answers is not in the document]
Comment (bridiver) [reply browser/brave_local_state_prefs.cc:144]: I guess at some point skus was converted to local state so I guess this is correct here, but some things should change in skus

[the remark this answers is not in the document]
Comment (nvonpentz) [reply browser/brave_local_state_prefs.cc:144]: Right, the intention was to be consistent with how it's done in the BraveVPNService https://github.com/brave/brave-core/blob/ec75711c56faca5146f18300b697cb34c1032375/components/brave_vpn/common/brave_vpn_utils.cc#L239.

File: components/ai_chat/common/pref_names.h (hunk 12 above)
Comment (bridiver) [reply components/ai_chat/common/pref_names.h:25]: does this need security review?

File: components/ai_chat/core/test/BUILD.gn [hunk 15]
```diff
@@ -0,0 +1,25 @@
+import("//brave/build/config.gni")
+import("//testing/test.gni")
+
+source_set("ai_chat_core_unit_tests") {
```
Comment (bridiver) [inline components/ai_chat/core/test/BUILD.gn:4]: this should not go in a separate directory, it should go in the same directory as the source files. Also the naming convention is just "unit_tests"

File: components/ai_chat/common/pref_names.h (hunk 12 above)
Comment (bridiver) [reply components/ai_chat/common/pref_names.h:25]: and does this really need to be stored in prefs vs just keeping them in-memory?

Comment (bridiver) [reply components/ai_chat/common/pref_names.h:25]: also if we're going to keep these, shouldn't it be more generic for all services and not have every service keep its own cache?

File: components/ai_chat/core/engine/remote_completion_client.cc [hunk 16]
```diff
@@ -139,7 +126,30 @@ void RemoteCompletionClient::QueryPrompt(
...
+    EngineConsumer::GenerationCompletedCallback data_completed_callback,
+    EngineConsumer::GenerationDataCallback data_received_callback,
+    absl::optional<CredentialCacheEntry> credential) {
+  bool premium_enabled = credential.has_value();
+  const GURL api_base_url = GetEndpointBaseUrl(premium_enabled);
+  base::flat_map<std::string, std::string> headers;
+  if (premium_enabled) {
+    // Add Leo premium SKU credential as a Cookie header.
+    std::string cookie_header_value =
+        "__Secure-sku#brave-leo-premium=" + credential->credential;
```
Comment (thypon) [inline components/ai_chat/core/engine/remote_completion_client.cc:148]: Isn't this a JSON API?
Shouldn't we rather use a `Authorization:` header or a specific `X-Brave-Premium:` header?

Comment (nvonpentz) [reply components/ai_chat/core/engine/remote_completion_client.cc:148]: I talked with @evq about this, we decided to send as a cookie rather than an Authorization header for consistency.  Currently the subscription service [accepts](https://github.com/brave-intl/subscriptions/blob/ad811bf4c8f3063bf4c0012f2d29f6f2f7b2d7b6/pkg/api/middleware.go#L132-L206) the credential as a cookie from the Client (which I believe is Guardian). So we're just doing the same here.

**Comment by petemill** [comment]
Every single time we call `credential_manager_->GetPremiumStatus(std::move(callback));` it performs an HTTP request. This results in a delay in the UI finding out if the user is currently premium or not every time a side panel is opened (not to mention that we have to call this a lot since we don't have any observable events, so this might be a waste of API resources?).

Should we cache this in the AI Chat layer, or is this something that should be cached in the CredentialManager, or even SKUs layer?

File: components/ai_chat/common/pref_names.h (hunk 12 above)
Comment (nvonpentz) [reply components/ai_chat/common/pref_names.h:25]: > does this need/already had security review?

Security review requested [here](https://github.com/brave/reviews/issues/1424)

> and does this really need to be stored in prefs vs just keeping them in-memory?

I opted to store in prefs because I believe that gives us better guarantees that users' credentials will not be lost. E.g. if the server is down, and the user tries to chat a few times and fails, if they close the browser, those credentials would be lost.

Also this is how it's done for VPN.

> also if we're going to keep these, shouldn't it be more generic for all services and not have every service keep its own cache?

That does make sense, but that's a bigger change than I'm comfortable making here.  This logic works for AI chat, may not work for VPN (I know there is an additional guardian credential in the mix there), and we've also discussed other solutions to this problem that involve changes to the rust SDK.  For now, I'd like to narrowly solve this problem at the expense of generalizing well

Comment (bridiver) [reply components/ai_chat/common/pref_names.h:25]: I talked to @thypon about this so I think they'll cover it in security review. They may actually ask you to remove this entirely, but I'll leave that part to them because they're covering the client/server interaction part

**Review by bridiver — APPROVED** [review]
config.js changes ok

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by petemill — APPROVED** [review]
Working well. We'll handle the `ActiveDisconnected` state in a follow-up. https://github.com/brave/brave-browser/issues/33670

As mentioned earlier, it'll be more efficient to store 1 credential manager for the entire global browser rather than 1 per Tab, but looks like we can do that in a follow-up refactor, and credentials are stored via Skus and not anything in this PR.

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

```
Brave | 1.61.34 Chromium: 118.0.5993.88 (Official Build) nightly (64-bit)
-- | --
Revision | 59b8b92f5f620463841656bf6f32fcdf07b7f12e
OS | Windows 10 Version 22H2 (Build 19045.3570)
```

File: components/ai_chat/core/ai_chat_credential_manager_unittest.cc [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+#include "brave/components/ai_chat/common/pref_names.h"
+#include "brave/components/ai_chat/core/ai_chat_credential_manager.h"
+#include "brave/components/skus/browser/pref_names.h"
+#include "brave/components/skus/browser/skus_context_impl.h"
+#include "brave/components/skus/browser/skus_service_impl.h"
+#include "brave/components/skus/browser/skus_utils.h"
+#include "brave/components/skus/common/features.h"
+#include "brave/components/skus/common/skus_sdk.mojom.h"
+#include "components/prefs/testing_pref_service.h"
+#include "content/public/test/browser_task_environment.h"
```
Comment (bridiver) [inline components/ai_chat/core/ai_chat_credential_manager_unittest.cc:21]: you can't use content deps in core
